lib/core/providers/notifications_freefood.dart

```diff
@@ -36,30 +36,30 @@ class FreeFoodDataProvider extends ChangeNotifier {
   late MessagesDataProvider _messageDataProvider;
 
   ///SERVICES
-  late FreeFoodService _freeFoodService;
+  late FreeFoodService _freeFoodService = FreeFoodService();
 
   void initializeValues() {
-    _messageToCount = new HashMap<String, int?>();
-    _messageToMaxCount = new HashMap<String, int?>();
+    _messageToCount = new HashMap<String, int>();
+    _messageToMaxCount = new HashMap<String, int>();
     _registeredEvents = [];
   }
 
   void removeId(String id) {
     _messageToCount.remove(id);
     _messageToMaxCount.remove(id);
-    _registeredEvents!.remove(id);
+    _registeredEvents.remove(id);
   }
 
   void parseMessages() {
     // initializeValues();
-    List<MessageElement?> messages = _messageDataProvider.messages!;
-    messages.forEach((m) async {
-      if (m!.audience != null &&
-          m.audience!.topics != null &&
-          (m.audience!.topics!.contains("freeFood") || m.audience!.topics!.contains("testFreeFood"))) {
```

Comment:
@morebytes May we plan to revert or discuss for 7.31? The `testFreeFood` condition is is required to fully test Free Food notifications.